internal/service/advancedclustertpf/resource.go

```diff
@@ -99,80 +96,230 @@ func (r *rs) Update(ctx context.Context, req resource.UpdateRequest, resp *resou
+	}
+	_, summary, detail := r.awaitChanges(ctx, &state.Timeouts, diags, projectID, clusterName, "delete")
+	if summary != "" || detail != "" {
+		diags.AddError(summary, detail)
+	}
 }
 
 func (r *rs) ImportState(ctx context.Context, req resource.ImportStateRequest, resp *resource.ImportStateResponse) {
 	conversion.ImportStateProjectIDClusterName(ctx, req, resp, "project_id", "name")
 }
 
-func mockedSDK(ctx context.Context, diags *diag.Diagnostics, timeout timeouts.Value) (*TFModel, bool) {
-	sdkResp, err := ReadClusterResponse()
+func (r *rs) createCluster(ctx context.Context, plan *TFModel, diags *diag.Diagnostics, resp *resource.CreateResponse) (summary, detail string) {
+	sdkReq := NewAtlasReq(ctx, plan, diags)
+	if diags.HasError() {
+		return "", ""
+	}
+	api := r.Client.AtlasV220240805.ClustersApi
+	apiLegacy := r.Client.AtlasV220240530.ClustersApi
+	projectID := plan.ProjectID.ValueString()
+	clusterName := plan.Name.ValueString()
+	_, _, err := api.CreateCluster(ctx, projectID, sdkReq).Execute()
+	if err != nil {
+		return "errorCreate", fmt.Sprintf(errorCreate, err.Error())
+	}
+	cluster, summary, detail := r.awaitChanges(ctx, &plan.Timeouts, diags, projectID, clusterName, "create")
+	if summary != "" || detail != "" {
+		return summary, detail
+	}
+
+	var legacyAdvConfig *admin20240530.ClusterDescriptionProcessArgs
+	legacyAdvConfigUpdate := NewAtlasReqAdvancedConfigurationLegacy(ctx, &plan.AdvancedConfiguration, diags)
+	if legacyAdvConfigUpdate != nil {
+		legacyAdvConfig, _, err = apiLegacy.UpdateClusterAdvancedConfiguration(ctx, projectID, clusterName, legacyAdvConfigUpdate).Execute()
+		if err != nil {
+			// Maybe should be warning instead of error to avoid having to re-create the cluster
+			return "errorUpdateeAdvConfigLegacy", fmt.Sprintf(errorCreate, err.Error())
+		}
+	}
+
+	advConfigUpdate := NewAtlasReqAdvancedConfiguration(ctx, &plan.AdvancedConfiguration, diags)
+	var advConfig *admin20240805.ClusterDescriptionProcessArgs20240805
+	if advConfigUpdate != nil {
+		advConfig, _, err = api.UpdateClusterAdvancedConfiguration(ctx, projectID, clusterName, advConfigUpdate).Execute()
+		if err != nil {
+			// Maybe should be warning instead of error to avoid having to re-create the cluster
+			return "errorUpdateAdvConfig", fmt.Sprintf(errorCreate, err.Error())
+		}
+	}
+	return r.convertClusterAddAdvConfig(ctx, legacyAdvConfig, advConfig, cluster, plan.Timeouts, diags, &resp.State)
+}
+
+func (r *rs) readCluster(ctx context.Context, model *TFModel, state *tfsdk.State, diags *diag.Diagnostics) {
+	clusterName := model.Name.ValueString()
+	projectID := model.ProjectID.ValueString()
+	api := r.Client.AtlasV220240805.ClustersApi
+	readResp, _, err := api.GetCluster(ctx, projectID, clusterName).Execute()
 	if err != nil {
-		diags.AddError("errorCreate", fmt.Sprintf(errorCreate, err.Error()))
-		return nil, true
+		if admin.IsErrorCode(err, ErrorCodeClusterNotFound) {
+			state.RemoveResource(ctx)
```

Comment:
RemoveResource should only be called from Read,  readCluster is also called from Update: https://developer.hashicorp.com/terraform/plugin/framework/resources/update

```
Caveats:
An error is returned if the response state has the RemoveResource() method called. This method is not valid during update. Return an error if the resource is no longer exists.
```

Reply:
Good catch. I believe it is a low chance since read always should be called before update... Fixed with allowNotFOund